sdk/storage/src/common/StorageSharedKeyCredential.h

```diff
@@ -0,0 +1,26 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+#pragma once
+
+#include <mutex>
+#include <string>
+
+namespace Azure { namespace Storage {
+
+class StorageSharedKeyCredential
+{
+public:
+    StorageSharedKeyCredential(std::string accountName, std::string accountKey);
+
+    const std::string& GetAccountName() const;
```

Comment:
It doesn't make sense to return references to a thing which is supposed to be synchronized. Either the mutex should go or these need to return copies.